src/credential/issuance/03-start-credential-issuance.ts

```diff
@@ -323,25 +316,49 @@ export const startCredentialIssuance: StartCredentialIssuance = async (
 
 /**
  * Authorizes the user using the query mode and the authorization context.
- * The authorization context catches the 302 redirect from the authorization server which contains the authorization response.
  * @param authzRequestEndpoint The authorization endpoint of the authorization server
  * @param params The query parameters to be used in the request
- * @param redirectSchema The schema to be used in the redirect
- * @param authorizationContext The context to identify the user which will be used to start the authorization
+ * @param redirectUri The URL to which the redirect is made is usually a custom URL or deeplink
+ * @param authorizationContext The AuthorizationContext to manage the internal webview. If not specified, the default browser is used
  * @returns The authrozation result containing the authorization code, state and issuer
  */
 const authorizeUserWithQueryMode = async (
   authzRequestEndpoint: string,
   params: URLSearchParams,
-  redirectSchema: string,
-  authorizationContext: AuthorizationContext
+  redirectUri: string,
+  authorizationContext?: AuthorizationContext
 ): Promise<AuthorizationResult> => {
-  const authRedirectUrl = await authorizationContext
-    .authorize(`${authzRequestEndpoint}?${params}`, redirectSchema)
-    .catch((e) => {
-      throw new IdentificationError(e.message);
+  const authUrl = `${authzRequestEndpoint}?${params}`;
+  var authRedirectUrl: string | undefined;
+
+  if (authorizationContext) {
+    const redirectSchema = new URL(redirectUri).protocol.replace(":", "");
+    authRedirectUrl = await authorizationContext
+      .authorize(authUrl, redirectSchema)
+      .catch((e) => {
+        throw new IdentificationError(e.message);
+      });
+  } else {
+    // handler for redirectUri
+    Linking.addEventListener("url", ({ url }) => {
+      if (url.includes(redirectUri)) {
+        authRedirectUrl = url;
+      }
     });
 
+    Linking.openURL(authUrl);
+
+    /*
+     * Waits for 120 seconds for the identificationRedirectUrl variable to be set
+     * by the custom url handler. If the timeout is exceeded, throw an exception
+     */
+    await until(() => authRedirectUrl !== undefined, 120);
```

Comment:
Should we await both of them with a `Promise.all` which rejects as soon as the first one rejects?

Reply:
Thanks, fixed in [d5859f7](https://github.com/pagopa/io-react-native-wallet/pull/96/commits/d5859f7f7679a5651429918b040d8658196ae43e)